WebApi/WebApiTest/ControllerTest/PostControllerTest.cs

```diff
@@ -0,0 +1,279 @@
+using WebApi.Models.POCO;
+using Microsoft.Extensions.Logging;
+using WebApi.Services;
+using WebApi.Services.Serives_Implementations;
+using WebApi.Services.Services_Interfaces;
+using WebApi.Database;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using Moq;
+using WebApi.Models.DTO;
+using System.Threading.Tasks;
+using System.Security.Cryptography;
+using System.Linq;
+using NuGet.Frameworks;
+using Microsoft.VisualBasic;
+using WebApi.Database.Mapper;
+using Microsoft.AspNetCore.Mvc;
+using System.Linq.Expressions;
+using WebApi.Models.DTO.PostDTOs;
+
+namespace WebApiTest.ControllerTest
+{
+    public class PostControllerTest
+    {
+        int userID = 1;
+        DateTime in_time = DateTime.Now;
+
+        [Theory]
+        [InlineData(0, "Konrad Gaweda", 1, "RTV", "MyTitle", "Content", 41, false, false)]
+        [InlineData(1, "Jan Kowalski", 11, "RTV", "MyTitle1", "Conte nt1 ", 11, false, true)]
+        [InlineData(4, "Jan Gawęda", 1, "RTV", "MyTitle2", " Co ntent2 ", 33, true, false)]
+        [InlineData(7, "Konrad Kowalski", 32, "RTV", "MyTitle3", "Conten t3", 441, true, true)]
+        public void GetAll_Test(int in_id, string in_author, int in_authorID, string in_category,
+            string in_title, string in_content, int in_likesCount, bool in_isLiked, bool in_isPromoted)
+        {
+            List<PostDTO> posts = new List<PostDTO>();
+            posts.Add(new PostDTO
+            {
+                id = in_id,
+                author = in_author,
+                authorID = in_authorID,
+                category = in_category,
+                title = in_title,
+                content = in_content,
+                likesCount = in_likesCount,
+                datetime = in_time,
+                isLikedByUser = in_isLiked,
+                isPromoted = in_isPromoted
+            });
+
+            posts.Add(new PostDTO
+            {
+                id = in_id + 1,
+                author = in_author + " Second",
+                authorID = in_authorID + 1,
+                category = in_category + "321",
+                title = in_title+ " One",
+                content = in_content + " Three",
+                likesCount = in_likesCount + 14,
+                datetime = DateTime.Now,
+                isLikedByUser = in_isLiked = !in_isLiked,
+                isPromoted = in_isPromoted = ! in_isPromoted
+            });
+
+            //Arrange
+            var mockService = new Mock<IPostService>();
+            mockService.Setup(x => x.GetAll()).Returns(posts.AsQueryable());
+            var mockLogger = new Mock<ILogger<PostController>>();
+            var controller = new PostController(mockLogger.Object, mockService.Object);
+
+            var expected = posts;
+            //Act
+            var actual = controller.GetAll(userID).ToList();
+
+            //Asset
+            Assert.True(expected.All(shouldItem => actual.Any(isItem => isItem == shouldItem)));
+        }
```

Comment:
Skoro do GetAll w kontrolerze dajesz userID to w jaki sposób getujesz all tym serwisem bez przekazania userID? Chyba że getujesz ALL i już w kontrolerze filtrujesz (ale po kodzie który widziałem nie wydaje mi się).

Reply:
Dobra widzę że masz testy pod GetAll z userID oraz bez. Czyli po prostu błąd przy kopiowaniu jeśli dobrze rozumiem

Reply:
Po prostu są 2 metody. Jedna pobiera wszystkie posty jakie istnieją a druga wszystkie posty danego użytkownika.
Pierwsza przyjmuje sobie userID, które jest weryfikacją czy informacje chce pobrać admin(jak dla mnie głupota).
Druga przyjmuje userID, którego posty ma pobrać z endpointa, więc to nie pomyłka przy kopiowaniu.